staging/src/k8s.io/controller-manager/pkg/clientbuilder/client_builder_dynamic.go

```diff
@@ -156,6 +162,37 @@ func (t *DynamicControllerClientBuilder) ClientOrDie(name string) clientset.Inte
 	return client
 }
 
+type resettableTokenSource interface {
+	oauth2.TokenSource
+	ResetToken()
+}
+
+type resettableTokenSourceTransport struct {
+	base http.RoundTripper
+	src  resettableTokenSource
+}
+
+func (rt *resettableTokenSourceTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	resp, err := rt.base.RoundTrip(req)
+	if apierrors.IsUnauthorized(err) {
+		rt.src.ResetToken()
```

Comment:
would it make sense to wait for more than one unauthorized error in a row before force-refreshing the token?

Reply:
if the token failed once i'd expect it to fail twice because it is cached.

Reply:
@deads2k - thoughts here on threshold for resetting the token (e.g. number of consecutive 401s, or time elapsed between first 401)?